src/components/BurgerConstructor/BurgerConstructor.tsx

```diff
@@ -25,15 +25,37 @@ function BurgerConstructor() {
   const {user} = useAppSelector(state => state.auth);
   const {orderRequest} = useAppSelector(state => state.order)
   const [modal, setModal] = React.useState(false);
-  const openModal = () => {
-    if (!orderRequest && user.email) {
-      const authToken = getCookie('token');
-      dispatch(postOrder(stuffing !== null ? [bun!._id, ...stuffing!.map(item => item._id)] : [bun!._id], authToken!));
-      dispatch(clearConstructor());
-    } else {
+  // const openModal = () => {
+  //   const authToken = getCookie('token');
+  //   if (!orderRequest && user.email && bun && authToken) {
+  //     dispatch(postOrder(stuffing !== null ? [bun._id, ...stuffing.map(item => item._id)] : [bun._id], authToken));
+  //     dispatch(clearConstructor());
+  //   } else {
+  //     navigate(ROUTES.login)
+  //   }
+  //   setModal(true)
+  // };
```

Comment:
If you want to keep this, you need to leave a comment with the reason - for your future self and anyone reading the code. Otherwise, remove it.